src/transformers/file_utils.py

```diff
@@ -297,6 +297,19 @@ def is_pandas_available():
     return importlib.util.find_spec("pandas") is not None
 
 
+def is_sagemaker_distributed_available():
+    # Get the sagemaker specific env variable.
+    sagemaker_params = os.getenv("SM_FRAMEWORK_PARAMS", "")
+    if len(sagemaker_params) == 0:
```

Comment:
When I run the `run_glue.py` on a single-GPU machine it fails. 
`len(os.environ['SM_FRAMEWORK_PARAMS']) === 2`. `{}` is not interpreted as an empty `dict` it is an `string`. 
We could either compare `if len(sagemaker_params) > 2` move the `json.loads(sagemaker_params)` above and the compare with `if len(sagemaker_params) == 0`

Reply:
Oh in that case I can set the default to `"{}"` and wrap the `json.loads` in a try/except block for a more robust test.